lib/AST/Decl.cpp

```diff
@@ -1901,18 +1902,16 @@ Accessibility ValueDecl::getFormalAccessImpl(const DeclContext *useDC) const {
   return getFormalAccess();
 }
 
-const DeclContext *
-ValueDecl::getFormalAccessScope(const DeclContext *useDC) const {
+AccessScope ValueDecl::getFormalAccessScope(const DeclContext *useDC) const {
   const DeclContext *result = getDeclContext();
   Accessibility access = getFormalAccess(useDC);
 
   while (!result->isModuleScopeContext()) {
     if (result->isLocalContext())
-      return result;
+      return AccessScope(result);
```

Comment:
I think this one should be considered "private" as well. Maybe the extra flag should be to mark file scopes fileprivate instead?